### PR TITLE
Refactor the test suite

### PR DESCRIPTION
**Pull request by j-f1**
This adds a test helper that creates a testing function, which is usually called `thePlugin`, which streamlines testing transforms. I’ve also gone through and edited some of the test names for clarity.

### CODE REVIEW

File: utils/test-transform.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        });
+      } catch (e) {
+        // use the stack from outside the it() clause
+        // (the one inside the clause doesn’t show the actual test code)
+        e.stack = stack;
+        throw e;
+      }
+    });
+  };
+  thePlugin.skip = name => it.skip(name);
```
Comment (vigneshshanmugam) [inline utils/test-transform.js]: you would also need to support `only` if we would like to run a single test during dev..

**Comment by vigneshshanmugam** [comment]
The cleanup looks good overall.. Thanks a lot 👍

File: utils/test-transform.js (hunk 1 above)
Comment (j-f1) [inline utils/test-transform.js]: Done!

**Comment by boopathi** [comment]
This looks great!

I thought of moving it to fixtures based testing - each test case in 2 files - `source.js` and `expected.js`. But this is fine for now.

**Pull request merged**